functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -351,6 +351,34 @@ oneOutput cross_batch_rule(const Tensor& self, c10::optional<int64_t> self_bdim,
   return std::make_tuple(linalg_cross(self_, other_, dim_), 0);
 }
 
+c10::optional<int64_t> batch_dim_if_not_empty(const Tensor& t) {
+  if (t.dim() == 1 && t.size(0) == 0) {
+    return c10::optional<int64_t>();
+  }
+  return c10::optional<int64_t>(0);
+}
+
+fourOutputs linalg_lstsq_batch_rule(
+    const Tensor& self, c10::optional<int64_t> self_bdim, const Tensor& b, c10::optional<int64_t> b_bdim,
+    c10::optional<double> rcond, c10::optional<c10::string_view> driver) {
+  TORCH_CHECK(!self_bdim.has_value() || self.dim() > 2, "torch.linalg.lstsq: input must have at least 2 dimensions.");
+  TORCH_CHECK(!b_bdim.has_value() || b.dim() > 1, "torch.linalg.lstsq: other must have at least 1 dimension.");
+
+  const auto batch_size = get_bdim_size2(self, self_bdim, b, b_bdim);
+  const auto tensor_other = _binary_pointwise_helper(self, self_bdim, b, b_bdim, false);
```

Comment:
nit: when passing in boolean flags, putting the argument name in a comment makes it easier to read. Something like `_binary_pointwise_helper(self, self_bdim, b, b_bdim, /*do_type_promotion*/false)` (I don't actually remember what the flag is)